txconnpool/pool.py

```diff
@@ -62,14 +65,17 @@ def clientConnectionLost(self, connector, reason):
             connector,
             reason)
 
-
     def clientConnectionFailed(self, connector, reason):
         """
         Notify the connectionPool that we're unable to connect
         """
+
         if self._protocolInstance is not None:
             self.connectionPool.clientBusy(self._protocolInstance)
 
+        self.connectionPool.clientFailed(
+            self._protocolInstance, connector.host, connector.port)
```

Comment:
/nit

`connector` has a [`getDestination`](https://github.com/twisted/twisted/blob/cdf6441ea0b46b2b65c31be55c3f651f920ab9c2/twisted/internet/tcp.py#L1180) method that returns IPAdress. So maybe we can just pass connector to `clientFailed` and `clientFailed` can call `connector.getDestination()`